python-sdk/src/astro/sql/operators/base_decorator.py

```diff
@@ -108,6 +161,25 @@ def execute(self, context: Context) -> None:
         # if there is no SQL to run we raise an error
         if self.sql == "" or not self.sql:
             raise AirflowException("There's no SQL to run")
+        return context
+
+    def render_template_fields(
+        self,
+        context: Context,
+        jinja_env: jinja2.Environment | None = None,
+    ) -> BaseOperator | None:
+        """Template all attributes listed in template_fields.
+
+        This mutates the attributes in-place and is irreversible.
+
+        :param context: Dict with values to apply on content
+        :param jinja_env: Jinja environment
+        """
+        context = self._enrich_context(context)
+        return super().render_template_fields(context, jinja_env)
+
+    def execute(self, context: Context) -> None:
+        self._enrich_context(context)
```

Comment:
```suggestion
        context = self._enrich_context(context)
```
Shouldn't we also save this here?